### ISSUE TITLE
2 - In S3 mode, provide AWS ETag to the uploadSuccess callback when available

### ISSUE THREAD

**Issue by paulmelnikow** [FineUploader/fine-uploader#1118]
Hi there, and thanks for the great library!

I'm working on [drf-to-s3](https://github.com/bodylabs/drf-to-s3), a server component for direct-to-S3 uploads based on Django REST Framework and designed for Fine Uploader.

My goal is to make it as difficult as possible for a malicious user to substitute her/his own content for another user's upload.

I'm handling this by prefixing the upload key with the username, and refusing to sign policy documents outside the user's namespace. This works okay, but requires integrators to configure `objectProperties.key`, and also requires that the front end have access to that username. It also makes anonymous uploads tricky to handle.

A more elegant solution would be to compare the object's ETag to the ETag provided at the end of the upload, to validate that no one has tampered with the contents.

Since I have users upload to a temporary bucket and then move the file to a storage bucket, I can use the `x-amz-copy-source-if-match` header to a copy request, which atomically will validate the etag and copy the file.

With Fine Uploader's recommended CORS policy, AWS sets the ETag header on the POST response, so there's shouldn't be a need to calculate anything on the client.

Basically I ask that, if the ETag is present in the POST response, you would please include it with the form data in the uploadSuccess callback.
